llvm/include/llvm/Support/ThreadPool.h

```diff
@@ -209,25 +231,66 @@ class ThreadPool {
   /// Number of threads active for tasks in the given group (only non-zero).
   DenseMap<ThreadPoolTaskGroup *, unsigned> ActiveGroups;
 
-#if LLVM_ENABLE_THREADS // avoids warning for unused variable
   /// Signal for the destruction of the pool, asking thread to exit.
   bool EnableFlag = true;
-#endif
 
   const ThreadPoolStrategy Strategy;
 
   /// Maximum number of threads to potentially grow this pool to.
   const unsigned MaxThreadCount;
 };
 
+/// A non-threaded implementation.
+class SingleThreadExecutor : public ThreadPoolInterface {
```

Comment:
`#if !LLVM_ENABLE_THREADS`. But maybe if you didn't do it, you want this code compiled at all times?

Reply:
I was trying to use as little as possible so that the code is always at least "parsed" and we are less likely to break it (and also my IDE does not gray this entirely as a comment because of the macro).

But this is a weak argument, happy to restore a big `#if LLVM_ENABLE_THREADS` `#else` around the two classes.

Reply:
In that case, do you think we can add or extend some of the existing unit tests, to exercise the `SingleThreadExecutor` explicitly?

Reply:
I was gonna say, it happened several times in the past that the build was broken in `!LLVM_ENABLE_THREADS` configurations, which not many people are testing.

Reply:
I just implemented it: now the unit-tests are ran for both the single thread and std::thread implementation (except when LLVM_ENABLE_THREADS=OFF, then only the single-thread implementation is tested).

Can you take a look?